### DISCUSSION TITLE
OpenAPI documentation for admin/manager Api

### DISCUSSION THREAD

**Discussion by MekelCon** [apache/apisix#9406] [Ideas]
Hello !

I am wondering if it exist any existing initiative about providing an OpenApi Specification to describe the existing API of Apisix.
In my case i wanted to generate a client for spingboot using https://openapi-generator.tech/ to manage my apisix with some business logic before.
I found some old closed issue [here](https://github.com/apache/apisix/issues/1146) but found nothing since.

Is this something planned ? 
If i provide a valid swagger at the state of today (regarding Path and Model definition) what about his maintenance ? 
A good start point is what exist in [Schema defintion lua](https://github.com/apache/apisix/blob/master/apisix/schema_def.lua) but then we will have 2 files to maintain : 
-  the schema_def.lua
- The Swagger

Having a complete OAS swagger many advantage : 
   - Provide a documentation in a very common and standard way
   - Allow to use other tool for code generation, client generation
   - ...

Thank you for your help 👍

**Comment by Lensual** [comment]
here

https://github.com/apache/apisix/issues/10892#issuecomment-1919047887
